### PR TITLE
[enhancement](Nereids): avoid contruct groupExpr in graph-simplifier

### PR DESCRIPTION
**Pull request by keanji-x**
Signed-off-by: xiejiann <jianxie0@gmail.com>

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.34 seconds
 load time: 465 seconds
 storage size: 17171000255 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230206065031_clickbench_pr_90710.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/joinorder/hypergraph/GraphSimplifier.java [hunk 1]
```diff
@@ -416,35 +422,32 @@ private boolean tryGetSuperset(long bitmap1, long bitmap2, List<Long> superset)
...
-        inputs.add(PhysicalProperties.ANY);
-        groupExpression.updateLowestCostTable(PhysicalProperties.ANY, inputs, cost);
-
-        return newJoin.withGroupExpression(Optional.of(groupExpression));
+    private double calCost(Edge edge, StatsDeriveResult stats,
+            StatsDeriveResult leftStats, StatsDeriveResult rightStats) {
+        LogicalJoin join = edge.getJoin();
+        PlanContext planContext = new PlanContext(stats, leftStats, rightStats);
+        double cost = 0;
+        if (JoinUtils.shouldNestedLoopJoin(join)) {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/joinorder/hypergraph/GraphSimplifier.java:430]: looks like should swap the "if else" logic?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java [hunk 2]
```diff
@@ -99,6 +99,8 @@
  */
 public class StatsCalculator extends DefaultPlanVisitor<StatsDeriveResult, Void> {
 
+    public static final StatsCalculator STATS_CALCULATOR = new StatsCalculator(null);
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java:102]: what's STATS_CALCULATOR used for?

Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java:102]: Calculate cost for plan without groupexpression

**Review by starocean999 — APPROVED** [review]

**Pull request merged**